### PR TITLE
Allow setting your own CA as a kube secret

### PR DESCRIPTION
**Pull request by ishustava**
This PR adds the following values to the Helm chart to allow bringing your CA as a Kubernetes secret:
```yaml
tls:    
  caCert:
    secretName: ""
    secretKey: tls.crt
  caKey:
    secretName: ""
    secretKey: tls.key
```

It assumes the defaults for secret keys to follow the pattern of the Kubernetes TLS secrets. This also significantly simplifies the underlying implementation in that we don't need to check if the key is set and can instead assume it's always set.

### CODE REVIEW

File: templates/client-daemonset.yaml
Comment (lkysow) [inline templates/client-daemonset.yaml]: If we do
```yaml
  volumes:
    - name: tls-ca-cert
      secret:
        {{- if .Values.global.tls.caCert.secretName }}
        secretName: {{ .Values.global.tls.caCert.secretName }}
        {{- else }}
        secretName: {{ template "consul.fullname" . }}-ca-cert
        {{- end }}
        items:
        - key: {{ default "tls.crt" .Values.global.tls.caCert.secretKey }}
          path: tls.crt
```

Then the ca cert will always end up in the file `tls.crt`. What do you think about that pattern instead of defaulting the `secretKey` to `tls.crt` in `values.yaml`? I think I like it better because the file system always looks the same, regardless of how it's being initialized.

Comment (ishustava) [reply templates/client-daemonset.yaml]: I really like this idea, thanks! I've updated everything to use this pattern instead.

File: templates/client-daemonset.yaml
Comment (lkysow) [inline templates/client-daemonset.yaml]: I think this is now unnecessary with the changes you made for `path`.

File: templates/tests/test-runner.yaml
Comment (lkysow) [inline templates/tests/test-runner.yaml]: not needed?

File: values.yaml
Comment (lkysow) [inline values.yaml]:
```suggestion
    # of the CA to use for TLS communication within the Consul cluster.
```

File: values.yaml
Comment (lkysow) [inline values.yaml]:
```suggestion
    # caCert is a Kubernetes secret containing the certificate
```

File: values.yaml
Comment (lkysow) [inline values.yaml]: We've been using `null` for these in other places

**Review by lkysow — APPROVED** [review]
Looks good. We should wait until after namespaces to merge. I've been testing this with the wan federation work.

File: values.yaml
Comment (ishustava) [reply values.yaml]: I think I looked at gossip encryption which happens to be the only one that uses empty strings 😄 . It seems like `null` is more common; I'll update.

**Pull request merged**
